plugins/PostProcessingPlugin/scripts/ChituMods.py

```diff
@@ -0,0 +1,189 @@
+# Cura PostProcessingPlugin
+# Author:   Daniuel Spannbauer
+# Date:     November 3, 2019
+
+# Description:  This plugin generates and inserts code including a image of the
+#               slices part.
+
+
+from ..Script import Script
+from cura.Snapshot import Snapshot
+from UM.Logger import Logger
+import re
+
+def getValue(line, key, default=None):
+        if key not in line:
+            return default
+        else:
+            subPart = line[line.find(key) + len(key):]
+            m = re.search('^-?[0-9]+\\.?[0-9]*', subPart)
+            #if m is None:
+            #    pass
+            #return default
+        try:
+            return float(m.group(0))
+        except:
+            return default
+
+class ChituMods(Script):
+    def __init__(self):
+        super().__init__()
+        self._snapshot = None
+
+    def getSettingDataString(self):
+        return """{
+            "name": "Insert mods for chitu boards",
+            "key": "ChituMods",
+            "metadata": {},
+            "version": 2,
+            "settings":
+            {
+                "insert_preview_image":
+                {
+                    "label": "Insert a preview image for printers with chitu boards?",
+                    "description": "Selecting this a image will be generated and included into the gcode so the touch displays can show it",
+                    "type": "bool",
+                    "default_value": false
+                },
+                "insert_time_info":
+                
+                {
+                    "label": "Insert the needed time and the elapsed time?",
+                    "description": "Selecting this the calculated time and the elapsed time will be included in chitu style",
+                    "type": "bool",
+                    "default_value": false
+                }
+            }
+        }"""
+
+    def execute(self, in_data):
+        # we get a list, each list item is the command set for a complete layer
+        self.out_data = in_data

```

Comment:
There is no need to keep the out_data beyond the scope of this execute method. Instead of `self.out_data` you can just use `out_data`.

Reply:
You are right. Thanks.

Done